Clang-Tidy: Issues with Windows paths with forward slashes are not supported

The Clang-Tidy sensor is using the regex below:
```
((?>[a-zA-Z]:\\)??[^:]++):(\d{1,5}):(\d{1,5}): ([^:]++): (.+)
```
Issues with Windows paths with forward slashes are not supported:
```
D:/Path/Sample.cpp:78:23: warning: use std::make_unique instead [modernize-make-unique]
```

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -82,7 +82,7 @@ build_script:
   - C:\Python27\Scripts\pip.exe install requests
   - C:\Python27\Scripts\pip.exe install behave
   - C:\Python27\Scripts\pip.exe install colorama
-  - mvn versions:set -DnewVersion=2.0.4.%APPVEYOR_BUILD_NUMBER%
+  - mvn versions:set -DnewVersion=2.0.5.%APPVEYOR_BUILD_NUMBER%
   - mvn clean install -B -e -V
   - IF "%APPVEYOR_PULL_REQUEST_NUMBER%"=="" (mvn sonar:sonar -B -e -V -Dsonar.organization=sonaropencommunity -Dsonar.host.url=https://sonarcloud.io -Dsonar.login=%sonarcloudtoken%)
   - REM IF NOT "%APPVEYOR_PULL_REQUEST_NUMBER%"=="" (mvn sonar:sonar -B -e -X -V -Dsonar.verbose=true -Dsonar.pullrequest.base=master -Dsonar.pullrequest.branch=%APPVEYOR_PULL_REQUEST_HEAD_REPO_BRANCH% -Dsonar.pullrequest.key=%APPVEYOR_PULL_REQUEST_NUMBER% -Dsonar.pullrequest.provider=GitHub -Dsonar.pullrequest.github.repository=%APPVEYOR_PROJECT_SLUG%)

--- a/cxx-checks/pom.xml
+++ b/cxx-checks/pom.xml
@@ -5,7 +5,7 @@
   <parent>
     <groupId>org.sonarsource.sonarqube-plugins.cxx</groupId>
     <artifactId>cxx</artifactId>
-    <version>2.0.4-SNAPSHOT</version>
+    <version>2.0.5-SNAPSHOT</version>
   </parent>
 
   <artifactId>cxx-checks</artifactId>

--- a/cxx-sensors/pom.xml
+++ b/cxx-sensors/pom.xml
@@ -5,7 +5,7 @@
   <parent>
     <groupId>org.sonarsource.sonarqube-plugins.cxx</groupId>
     <artifactId>cxx</artifactId>
-    <version>2.0.4-SNAPSHOT</version>
+    <version>2.0.5-SNAPSHOT</version>
   </parent>
 
   <artifactId>cxx-sensors</artifactId>

--- a/cxx-sensors/src/main/java/org/sonar/cxx/sensors/clangtidy/ClangTidyParser.java
+++ b/cxx-sensors/src/main/java/org/sonar/cxx/sensors/clangtidy/ClangTidyParser.java
@@ -33,7 +33,7 @@ public class ClangTidyParser {
 
   private static final Logger LOG = Loggers.get(ClangTidyParser.class);
 
-  private static final String REGEX = "((?>[a-zA-Z]:\\\\)??[^:]++):(\\d{1,5}):(\\d{1,5}): ([^:]++): (.+)";
+  private static final String REGEX = "((?>[a-zA-Z]:[\\\\/])??[^:]++):(\\d{1,5}):(\\d{1,5}): ([^:]++): (.+)";
   private static final Pattern PATTERN = Pattern.compile(REGEX);
 
   private final CxxClangTidySensor sensor;

--- a/cxx-sensors/src/main/java/org/sonar/cxx/sensors/drmemory/DrMemoryParser.java
+++ b/cxx-sensors/src/main/java/org/sonar/cxx/sensors/drmemory/DrMemoryParser.java
@@ -37,7 +37,7 @@ public final class DrMemoryParser {
   private static final Logger LOG = Loggers.get(DrMemoryParser.class);
   private static final Pattern RX_MESSAGE_FINDER = Pattern.compile("^Error #\\d{1,6}:(.*)");
   private static final Pattern RX_FILE_FINDER = Pattern.compile(
-    "\\s*+#[^\\[]++\\[((?>[a-zA-Z]:\\\\)??[^:]++):(\\d{1,5})\\]");
+    "\\s*+#[^\\[]++\\[((?>[a-zA-Z]:[\\\\/])??[^:]++):(\\d{1,5})\\]");
   private static final int TOP_COUNT = 4;
 
   private DrMemoryParser() {

--- a/cxx-squid-bridge/pom.xml
+++ b/cxx-squid-bridge/pom.xml
@@ -5,7 +5,7 @@
   <parent>
     <groupId>org.sonarsource.sonarqube-plugins.cxx</groupId>
     <artifactId>cxx</artifactId>
-    <version>2.0.4-SNAPSHOT</version>
+    <version>2.0.5-SNAPSHOT</version>
   </parent>
 
   <artifactId>cxx-squid-bridge</artifactId>

--- a/cxx-squid/pom.xml
+++ b/cxx-squid/pom.xml
@@ -5,7 +5,7 @@
   <parent>
     <groupId>org.sonarsource.sonarqube-plugins.cxx</groupId>
     <artifactId>cxx</artifactId>
-    <version>2.0.4-SNAPSHOT</version>
+    <version>2.0.5-SNAPSHOT</version>
   </parent>
 
   <artifactId>cxx-squid</artifactId>

--- a/cxx-sslr-toolkit/pom.xml
+++ b/cxx-sslr-toolkit/pom.xml
@@ -4,7 +4,7 @@
   <parent>
     <groupId>org.sonarsource.sonarqube-plugins.cxx</groupId>
     <artifactId>cxx</artifactId>
-    <version>2.0.4-SNAPSHOT</version>
+    <version>2.0.5-SNAPSHOT</version>
   </parent>
 
   <artifactId>cxx-sslr-toolkit</artifactId>

--- a/integration-tests/pom.xml
+++ b/integration-tests/pom.xml
@@ -3,7 +3,7 @@
   <parent>
     <groupId>org.sonarsource.sonarqube-plugins.cxx</groupId>
     <artifactId>cxx</artifactId>
-    <version>2.0.4-SNAPSHOT</version>
+    <version>2.0.5-SNAPSHOT</version>
   </parent>
 
   <artifactId>integration-tests</artifactId>

--- a/pom.xml
+++ b/pom.xml
@@ -12,7 +12,7 @@
 
   <groupId>org.sonarsource.sonarqube-plugins.cxx</groupId>
   <artifactId>cxx</artifactId>
-  <version>2.0.4-SNAPSHOT</version>
+  <version>2.0.5-SNAPSHOT</version>
   <packaging>pom</packaging>
 
   <name>Cxx</name>

--- a/sonar-cxx-plugin/pom.xml
+++ b/sonar-cxx-plugin/pom.xml
@@ -5,7 +5,7 @@
   <parent>
     <groupId>org.sonarsource.sonarqube-plugins.cxx</groupId>
     <artifactId>cxx</artifactId>
-    <version>2.0.4-SNAPSHOT</version>
+    <version>2.0.5-SNAPSHOT</version>
   </parent>
 
   <artifactId>sonar-cxx-plugin</artifactId>

